packages/next/next-server/server/next-server.ts

```diff
@@ -507,7 +508,26 @@ export default class Server {
                   finished: true,
                 }
               } else {
-                ;(_req as any)._nextDidRewrite = true
+                // external rewrite, proxy it
+                if (parsedDestination.protocol) {
+                  const proxy = new Proxy({
+                    target: updatedDestination,
+                    changeOrigin: true,
+                    ignorePath: true,
```

Comment:
We need tests on how this property handles the path and sub-paths.

Reply:
Updated the test to show the expected behavior more clearly